ArticleTemplates/assets/js/bootstraps/campaign.js

```diff
@@ -0,0 +1,68 @@
+define([], function () {
+    'use strict';
+
+    const rootPath = 'https://code.api.nextgen.guardianapps.co.uk/formstack-campaign/submit'
+
+    function init() {
+      const campaign = document.querySelector('.campaign--snippet form');
+      if (campaign) {
+        initCampaign(campaign);
+      }
+    }
+
+    function initCampaign(campaign) {
+      if (!navigator.onLine) {
+        displayOfflineMessage(campaign);
+      }
+
+      window.addEventListener('online', hideOfflineMessage.bind(null, campaign));
+      window.addEventListener('offline', displayOfflineMessage.bind(null, campaign));
+
+      campaign.addEventListener('submit', function (e) {
+        e.preventDefault();
+        const data = new FormData(campaign);
+        const json = {};
+        for(let [key, value] of data.entries()) {
```

Comment:
I'd be wary about using destructuring assignments, I'm not sure older versions of android will support them.